### PR TITLE
Use junit-quickcheck to test ConstantFactory's long and BigInteger conversion code.

### LINKED ISSUE #83: Improve value conversion code tests
As well as possibly other parts that are suitable to this approach.

### PR DESCRIPTION
**Pull request by jvdb**
This will resolve #83.

I hesitated due to loss of JDK7 testing on Travis. But we'll lose that quickly anyway as we move toward Java8 with Metal itself. So seems a reasonable step.

### CODE REVIEW

**Comment by rdvdijk** [comment]
We might be able to check for JDK version inside these tests, and re-enable Java 7 support for all other tests...

File: core/src/test/java/io/parsingdata/metal/expression/value/ConstantFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            final long maxValue = BigInteger.valueOf(2).pow(i-1).longValue();
+            if ((maxValue + input) >= 0) {
+                return BigInteger.valueOf(input).add(BigInteger.valueOf(2*maxValue));
+            }
+        }
+        return BigInteger.valueOf(input).add(TWOS_DIFF);
+    }
+
+    @Property
+    public void bigIntegerConstant(@InRange(min = "-1000000000000000000", max = "1000000000000000000") final BigInteger input) {
```
Comment (rdvdijk) [inline core/src/test/java/io/parsingdata/metal/expression/value/ConstantFactoryTest.java]: Shouldn't we have the full range of sizes here as well? (e.g. `bigIntegerConstant8`, `bigIntegerConstant16`, etc.?)

File: core/src/test/java/io/parsingdata/metal/expression/value/ConstantFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    private void checkLongConstant(final long input) {
+        assertEquals(input, ConstantFactory.createFromNumeric(input, signed()).asNumeric().longValue());
+        if (input >= 0) {
+            assertEquals(input, ConstantFactory.createFromNumeric(input, enc()).asNumeric().longValue());
+        } else {
+            assertEquals(0, calculateUnsignedValue(input).compareTo(ConstantFactory.createFromNumeric(input, enc()).asNumeric()));
+        }
+    }
+
+    private BigInteger calculateUnsignedValue(final long input) {
```
Comment (rdvdijk) [inline core/src/test/java/io/parsingdata/metal/expression/value/ConstantFactoryTest.java]: This deserves a comment.

File: core/src/test/java/io/parsingdata/metal/expression/value/ConstantFactoryTest.java (hunk 1 above)
Comment (jvdb) [inline core/src/test/java/io/parsingdata/metal/expression/value/ConstantFactoryTest.java]: @rdvdijk `BigInteger` has arbitrary sizes so "full range" is not realistic :)

But I see your point. Combining is also possible -- turn the values in the other tests into `BigInteger`s and then perform that test. Then add two specific tests for larger numbers (both positive and negatives values >64 bits).

**Comment by rdvdijk** [comment]
How about we add this to the `ConstantFactoryTest` file, re-enable Java 7 and see if Travis agrees?

`assumeTrue()` documentation states:
> If called with an expression evaluating to false, the test will halt and be ignored.

```java
    @Before
    public void checkJDK() {
        final String specVersion = System.getProperty("java.specification.version");
        final String[] specVersionParts = specVersion.split("\\.");
        final int majorVersion = Integer.parseInt(specVersionParts[0]);
        final int minorVersion = Integer.parseInt(specVersionParts[1]);
        assumeTrue("Tests only run on Java >= 8 (java.specification.version is: " + specVersion + ")", majorVersion >= 1 && minorVersion >= 8);
    }
```

**Comment by jvdb** [comment]
We can try but I don't think it'll work -- the class references junit-quickcheck _before_ executing the code so it will give the same error. But like stated earlier -- we will lose 1.7 compatibility down the road anyway, so no point in investing heavily in keeping it.

**Comment by jvdb** [comment]
This issue has instead been addressed by #116.

**Pull request closed without merging**